native/ex_vl_convert/src/lib.rs

```diff
@@ -0,0 +1,348 @@
+use rustler::Atom;
+use rustler::Encoder;
+use rustler::Env;
+use rustler::NifTuple;
+use rustler::Term;
+
+use vl_convert_rs::converter::VgOpts;
+use vl_convert_rs::converter::VlOpts;
+use vl_convert_rs::VlConverter;
+use vl_convert_rs::VlVersion;
+
+// +-------------------------------------+
+// |        Rustler Return Types         |
+// +-------------------------------------+
+
+mod atoms {
+    rustler::atoms! {
+        ok,
+        error
+    }
+}
+
+#[derive(NifTuple)]
+struct StringResultTuple {
+    lhs: Atom,
+    rhs: String,
+}
+
+#[derive(NifTuple)]
+struct BinaryResultTuple {
+    lhs: Atom,
+    rhs: Vec<u8>,
+}
+
+enum Either<BinaryResultTuple, StringResultTuple> {
+    BinaryTuple(BinaryResultTuple),
+    StringTuple(StringResultTuple),
+}
+
+impl Encoder for Either<BinaryResultTuple, StringResultTuple> {
+    fn encode<'b>(&self, env: Env<'b>) -> Term<'b> {
+        use Either::{BinaryTuple, StringTuple};
+
+        return match self {
+            BinaryTuple(result) => result.encode(env),
+            StringTuple(result) => result.encode(env),
+        };
+    }
+}
+
+// +-------------------------------------+
+// |            Vega Functions           |
+// +-------------------------------------+
+
+#[rustler::nif(schedule = "DirtyCpu")]
+fn vega_to_svg(vega_spec: String) -> StringResultTuple {
```

Comment:
Perhaps we could simplify by using the `Result` struct, as in:

```suggestion
fn vega_to_svg(vega_spec: String) -> Result<String, String> {
```

Then we return `Ok(...)` or `Err(...)` and it should be converted to :ok/:error tuple automatically?

Reply:
I had no idea that Rustler automatically serializes `Result` 🤯. Good call out....this greatly simplified things!